File: 3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.java
class Solution {
    private int reverse(int num) {
        int r = 0;
        while (num > 0) {
            r = r * 10 + num % 10;
            num /= 10;
        }
        return r;
    }

    public int minMirrorPairDistance(int[] nums) {
        int n = nums.length;
        HashMap<Integer, Integer> map = new HashMap<>();
        int r = 0, ans = Integer.MAX_VALUE;

        for (; r < n; ++r) {
            if (map.containsKey(nums[r]))
                ans = Math.min(ans, r - map.get(nums[r]));

            int rev = reverse(nums[r]);
            map.put(rev, r);
        }
        /*
        for (Map.Entry<Integer, Integer> em : map.entrySet())
            System.out.println("Key : " + em.getKey() + ". Value : " + em.getValue());
        System.out.println(ans);
        */
        return ((ans == Integer.MAX_VALUE) ? -1 : ans);
    }
}

File: 3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.py
class Solution:
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        n = len(nums)
        ans = inf
        Map = {}
        r = 0

        def reverse(num : int) -> int:
            rev = 0
            while num:
                rev = 10 * rev + num % 10
                num //= 10
            
            return rev

        while r < n:
            if nums[r] in Map:
                ans = min(ans, r - Map[nums[r]])
            
            rev = reverse(nums[r])
            Map[rev] = r
            r += 1
        
        return -1 if ans == inf else ans